### PR TITLE
Fix todays sql error and remember which machine did a task

### PR DESCRIPTION
**Pull request by snadrus**
## Related Issues
1. Schema problem 
2. Slack comment that there's no way to figure out which machine did some task to be able to dive into its logs for errors. 

## Proposed Changes
- Schema fixed, but still needs a "DROP SCHEMA lotus CASCADE". 
- harmony_task_history now remembers which hostname + port (identifier) did some piece of work. 

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: cmd/lotus-provider/run.go [hunk 1]
```diff
@@ -204,7 +204,10 @@ var runCmd = &cli.Command{
 
 		sa, err := StorageAuth(cfg.Apis.StorageRPCSecret)
 		if err != nil {
-			return xerrors.Errorf("parsing Apis.StorageRPCSecret config: %w", err)
+			return xerrors.Errorf(`'%w' while parsing the config toml's 
+	[Apis]
+	StorageRPCSecret=%v
+Get it from the JSON documents in ~/.lotus-miner/keystore called .PrivateKey`, err, cfg.Apis.StorageRPCSecret)
```
Comment (magik6k) [inline cmd/lotus-provider/run.go:210]: Should link specifically `~/.lotusminer/keystore/MF2XI2BNNJ3XILLQOJUXMYLUMU`

File: lib/harmony/harmonydb/sql/20231113.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE harmony_task_history ADD COLUMN completed_by_host_and_port varchar(300) NOT NULL;
```
Comment (magik6k) [inline lib/harmony/harmonydb/sql/20231113.sql:1]: tbh given that we're not rolled out anywhere yet, I'd say that we can afford to modify the main schema files, we can just drop the lotus schema between test runs;

(Not a super strong opinion)

**Review by magik6k — APPROVED** [review]
2 nits, but looks good generally

**Comment by snadrus** [comment]
@Reiers This is merged, and we are ready to continue testing.

**Pull request merged**